core/lib/tee_verifier/src/lib.rs

```diff
@@ -0,0 +1,712 @@
+//! Tee verifier
+//!
+//! Verifies that a L1Batch has the expected root hash after
+//! executing the VM and verifying all the accessed memory slots by their
+//! merkle path.
+//!
+//! The crate/library shadows some structures to avoid polluting the global code
+//! base with a stable (De-)Serialize ABI.
+//! If additional components need to (De-)Serialize these structures, we can revisit this decision.
```

Comment:
I'm not sure why we're hesitant to add `Serialize` and `Deserialize` traits to the existing structures in the global codebase, even if this file is the only one using serialization for them. Are there any concerns about performance or stability with the existing code that relies on these structures? I just want to make sure we really need all those 342 lines of boilerplate code.

Reply:
relevant PR https://github.com/matter-labs/zksync-era/pull/1723

Reply:
@mm-zk following the discussion from the PR @1723 (esp. in light of [this comment](https://github.com/matter-labs/zksync-era/pull/1723#issuecomment-2063364354)): do you have any thoughts on the boilerplate code mentioned above?

Reply:
I'm missing some context here, but I also don't like that we create shadow structures like this.

My main opposition against 'just adding serialize/deserialize' everywhere, was that this add additional work to whomever is modifyign the code later (as they have to consider the impact of adding/removing the fields, backwards compabilitiy etc).

But if you actually DO store these objects serialized in a database, then yes, let's add the Serialize/Deserialize to the 'main' objects - and discuss the backwards compatibility. (for example - can I go ahead and rename stuff ? or would it then require some database migration on your side).

Reply:
To add some context: we need to serialize these structures to send them of the wire (we do not plan to persist them for now). Our TEE prover will run on Azure, since Google does not offer SGX.

Reply:
@mm-zk updated this PR with a fixup removing the shadow structs and adding serialize/deserialize https://github.com/matter-labs/zksync-era/pull/1860/commits/cd5c5aace3f129baab8c487f823e47d5d3422ef7

Reply:
@mm-zk please let me know, if you have any objections